packages/chain-bridge/src/adapters/LayerZeroAdapter.sol

```diff
@@ -170,6 +206,122 @@ contract LayerZeroAdapter is
                         EXTERNAL INTERFACE
     //////////////////////////////////////////////////////////////*/
 
+    /// @inheritdoc IAssetAdapter
+    /**
+     * @dev Transfers assets cross-chain using LayerZero OFT protocol.
+     *      Flow: pull tokens → approve OFT → send via OFT → refund excess native.
+     *      If message is provided, encodes as OFT compose for post-delivery execution.
+     * @param operationId Unique identifier for this transfer operation
+     * @param params Transfer parameters (asset, amount, destination, optional message)
+     */
+    function transferAsset(
+        bytes32 operationId,
+        BridgeTypes.ExecuteTransferParams calldata params,
+        BridgeTypes.BridgeOptions calldata
+    )
+        external
+        payable
+        onlyTrustedDestination(params.destinationChainId)
+        onlyRouter
+        nonReentrant
+    {
+        // 1. Validate and get OFT
+        address oft = _getOFTForAsset(params.asset, params.amount);
+
+        // 2. Pull tokens
+        IERC20(params.asset).safeTransferFrom(
+            msg.sender,
+            address(this),
+            params.amount
+        );
+
+        // 3. Get destination and cache EID
+        uint32 dstEid = _externalIdForChain(params.destinationChainId);
+        address dstAdapter = _getAdapterPeerOrRevert(params.destinationChainId);
+
+        // 4. Approve OFT
+        IERC20(params.asset).forceApprove(oft, params.amount);
```

Comment:
for a token that is an OFT (asset == oft) approval is not needed as per 
```
Developers can choose between direct implementations where the token contract itself handles minting/burning and adapter patterns (where an intermediary or mint authority lock/burns tokens on one chain and unlock or mint them on another). Both approaches maintain unified supply and allow seamless cross-chain transfers.
``` but I'd double check